Fix 404 JSON smoke test in generated app.test.js

compwright opened the pull request:
`json: true` does not set the proper `Accept` header, so we need to set it.

daffl commented:
Thank you for the pull request. What happens if you don't have this in? I just tested it with a newly generated app and even without the change `request.error` is exactly the object that I'd expect it to be and the test is passing.

compwright commented:
With feathersjs/cli@3.7.1, accepting all defaults:

```
$ mkdir test-app
$ cd test-app
$ feathers generate app
? Project name test-app
? Description 
? What folder should the source files live in? src
? Which package manager are you using (has to be installed globally)? npm
? What type of API are you making? REST, Realtime via Socket.io
   create package.json
   create config/default.json
   create LICENSE
   create public/favicon.ico
   create public/index.html
   create .editorconfig
   create .eslintrc.json
   create src/app.hooks.js
   create src/channels.js
   create src/index.js
   create src/logger.js
   create src/hooks/log.js
   create src/middleware/index.js
   create src/services/index.js
   create .gitignore
   create README.md
   create src/app.js
   create test/app.test.js
   create config/production.json
npm WARN deprecated istanbul-lib-hook@1.2.1: 1.2.0 should have been a major version bump
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN test-app@0.0.0 No repository field.
npm WARN test-app@0.0.0 No license field.

+ @feathersjs/feathers@3.1.6
+ @feathersjs/express@1.2.3
+ @feathersjs/express@1.2.3
+ @feathersjs/configuration@1.0.2
+ @feathersjs/errors@3.3.0
+ serve-favicon@2.5.0
+ compression@1.7.2
+ cors@2.8.4
+ winston@2.4.2
+ helmet@3.12.1
+ @feathersjs/socketio@3.2.0
added 226 packages in 17.801s
npm WARN test-app@0.0.0 No repository field.
npm WARN test-app@0.0.0 No license field.

+ request-promise@4.2.2
+ request@2.87.0
+ mocha@5.2.0
+ eslint@4.19.1
added 161 packages in 12.849s


   ╭─────────────────────────────────────╮
   │                                     │
   │   Update available 5.6.0 → 6.1.0    │
   │     Run npm i -g npm to update      │
   │                                     │
   ╰─────────────────────────────────────╯

$ npm test

> test-app@0.0.0 test /Users/jhill/Development/test-app
> npm run eslint && npm run mocha


> test-app@0.0.0 eslint /Users/jhill/Development/test-app
> eslint src/. test/. --config .eslintrc.json


> test-app@0.0.0 mocha /Users/jhill/Development/test-app
> mocha test/ --recursive --exit

/Users/jhill/Development/test-app/src/logger.js:7
  format: format.combine(
                 ^

TypeError: Cannot read property 'combine' of undefined
    at Object.<anonymous> (/Users/jhill/Development/test-app/src/logger.js:7:18)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/jhill/Development/test-app/src/app.js:6:16)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/jhill/Development/test-app/test/app.test.js:4:13)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at /Users/jhill/Development/test-app/node_modules/mocha/lib/mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/Users/jhill/Development/test-app/node_modules/mocha/lib/mocha.js:247:14)
    at Mocha.run (/Users/jhill/Development/test-app/node_modules/mocha/lib/mocha.js:576:10)
    at Object.<anonymous> (/Users/jhill/Development/test-app/node_modules/mocha/bin/_mocha:637:18)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! test-app@0.0.0 mocha: `mocha test/ --recursive --exit`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the test-app@0.0.0 mocha script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jhill/.npm/_logs/2018-06-16T18_34_42_156Z-debug.log
npm ERR! Test failed.  See above for more details.
```

It appears this error is due to feathers installing winston@2.4.2 instead of 3.0.0. Upgrading fixes that issue, and shows the original issue this PR seeks to fix:

```
$ npm test

> test-app@0.0.0 test /Users/jhill/Development/test-app
> npm run eslint && npm run mocha


> test-app@0.0.0 eslint /Users/jhill/Development/test-app
> eslint src/. test/. --config .eslintrc.json


> test-app@0.0.0 mocha /Users/jhill/Development/test-app
> mocha test/ --recursive --exit



  Feathers application tests
    1) starts and shows the index page
    404
      2) shows a 404 HTML page
      3) shows a 404 JSON error without stack trace


  0 passing (222ms)
  3 failing

  1) Feathers application tests
       starts and shows the index page:
     RequestError: Error: socket hang up
      at new RequestError (node_modules/request-promise-core/lib/errors.js:14:15)
      at Request.plumbing.callback (node_modules/request-promise-core/lib/plumbing.js:87:29)
      at Request.RP$callback [as _callback] (node_modules/request-promise-core/lib/plumbing.js:46:31)
      at self.callback (node_modules/request/request.js:185:22)
      at Request.onRequestError (node_modules/request/request.js:877:8)
      at Socket.socketOnEnd (_http_client.js:423:9)
      at endReadableNT (_stream_readable.js:1064:12)
      at _combinedTickCallback (internal/process/next_tick.js:138:11)
      at process._tickCallback (internal/process/next_tick.js:180:9)

  2) Feathers application tests
       404
         shows a 404 HTML page:
     AssertionError [ERR_ASSERTION]: undefined == 404
      at rp.catch.res (test/app.test.js:38:16)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:689:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)

  3) Feathers application tests
       404
         shows a 404 JSON error without stack trace:
     AssertionError [ERR_ASSERTION]: undefined == 404
      at rp.catch.res (test/app.test.js:48:16)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:689:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)



npm ERR! code ELIFECYCLE
npm ERR! errno 3
npm ERR! test-app@0.0.0 mocha: `mocha test/ --recursive --exit`
npm ERR! Exit status 3
npm ERR! 
npm ERR! Failed at the test-app@0.0.0 mocha script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jhill/.npm/_logs/2018-06-16T18_37_47_239Z-debug.log
npm ERR! Test failed.  See above for more details.
```

compwright commented:
Did something change in a patch version of FeathersJS that changed the default response type or something? The only thing I can see that is different is a few packages are slightly behind on the patch version.

daffl commented:
No, only the version of Winston changed. I just double checked and I always get 3.0 installed. Maybe try updating your version of npm (`npm i npm -g`).

daffl commented:
Thank you again for looking into this! The Winston issues should now be solved by #370 by forcing the installation of version 3 which should prevent this error so I think we can close this PR.
